Software/src/sensors/lsm6ds33.cpp

```diff
@@ -196,24 +196,24 @@ int spartan::LSM6DS33::powerOff() {
 // Polling functions
 
 // TODO(harrisoncassar): Update this `update` function to utilize `ERROR_*` enums defined in `globals.h`
-bool spartan::LSM6DS33::update() {
+int spartan::LSM6DS33::update() {
     if (m_status == STATUS_OFF)
-        return false; // Should be returning ERROR_INVALID_STATUS
+        return ERROR_INVALID_STATUS; // Should be returning ERROR_INVALID_STATUS
 
     if (hasNewData() == RESULT_FALSE)
         //std::cerr << "No new data" << std::endl;
-        return false;
+        return ERROR_NO_NEW_DATA;
```

Comment:
I think this should also be returning `RESULT_FALSE`; having no new data is not necessarily an error.

Reply:
That makes sense. I'll also get rid of the `ERROR_NO_NEW_DATA` code since I made it for the sole purpose of reflecting that there was no new data.